imap_processing/tests/ultra/unit/test_helio_pset.py

```diff
@@ -0,0 +1,120 @@
+from unittest import mock
+
+import numpy as np
+import pandas as pd
+import pytest
+import xarray as xr
+
+from imap_processing import imap_module_directory
+from imap_processing.cdf.utils import load_cdf
+from imap_processing.tests.conftest import _download_external_data
+from imap_processing.ultra.constants import UltraConstants
+from imap_processing.ultra.l1c.helio_pset import calculate_helio_pset
+
+TEST_PATH = imap_module_directory / "tests" / "ultra" / "data" / "l1"
+
+
+@pytest.mark.skip(reason="Long running test for validation purposes.")
+def test_validate_exposure_time_and_sensitivities(ancillary_files, deadtime_datasets):
+    """Validates exposure time and sensitivities for ebin 0."""
+    sens_filename = "SENS-IMAP_ULTRA_90-IMAP_DPS-HELIO-nside32-ebin0.csv"
+    exposure_filename = "Exposures-IMAP_ULTRA_90-IMAP_DPS-HELIO-nside32-ebin0.csv"
+    de_filename = "imap_ultra_l1b_90sensor-de_20000101-repoint00000_v000.cdf"
+    test_data = [
+        (sens_filename, "ultra/data/l1/"),
+        (exposure_filename, "ultra/data/l1/"),
+        (de_filename, "ultra/data/l1/"),
+    ]
+    _download_external_data(test_data)
+    l1b_de = TEST_PATH / de_filename
+    l1b_de = load_cdf(l1b_de)
+    sensitivities_ebin_0 = pd.read_csv(TEST_PATH / sens_filename)
+    exposure_factor_ebin_0 = pd.read_csv(TEST_PATH / exposure_filename)
+
+    test_deadtimes = (
+        pd.read_csv(TEST_PATH / "test_p0_ebin0_deadtimes.csv", header=None)
+        .to_numpy()
+        .squeeze()
+    )
+    npix = 12288  # nside 32
+    # Create a minimal dataset to pass to the function
+    dataset = xr.Dataset(
+        {
+            "spin_number": (["epoch"], np.array([1, 2, 3])),
+        }
+    )
+    dataset.attrs["Repointing"] = "repoint00000"
+
+    pointing_range_met = (472374890.0, 582378000.0)
+    # Create mock spin data that has 5525 nominal spins
+    # Create DataFrame
+    nspins = 5522
+    nominal_spin_seconds = 15.0
+    spin_data = pd.DataFrame(
+        {
+            "spin_start_met": np.linspace(
+                pointing_range_met[0], pointing_range_met[1], nspins
+            ),
+            "spin_period_sec": np.full(nspins, nominal_spin_seconds),
+            "spin_phase_valid": np.ones(nspins),
+            "spin_period_valid": np.ones(nspins),
+        }
+    )
```

Comment:
There is a utility function `use_fake_spin_data_for_time` and `generate_fake_spin_data` that youmight be able to use here?